crates/storage/src/blueprint/sparse.rs

```diff
@@ -484,237 +484,483 @@ where
     }
 }
 
-/// The macro that generates SMT storage tests for the table with [`crate::structured_storage::test::InMemoryStorage`].
 #[cfg(feature = "test-helpers")]
-#[macro_export]
-macro_rules! root_storage_tests {
-    ($table:ident, $metadata_table:ident, $current_key:expr, $foreign_key:expr, $generate_key:ident, $generate_value:ident) => {
-        paste::item! {
-        #[cfg(test)]
-        mod [< $table:snake _root_tests >] {
-            use super::*;
-            use $crate::{
-                structured_storage::test::InMemoryStorage,
-                transactional::WriteTransaction,
-                StorageAsMut,
-            };
-            use $crate::rand::{
-                rngs::StdRng,
-                SeedableRng,
+#[allow(warnings)]
+/// Test module for root storage tests.
+pub mod root_storage_tests {
```

Comment:
if this is only for smt, then we should include it in the name of the mod

Reply:
Sure, though in general I don't think sub modules need to include the name of the parent module since that information is already stored in the namespace `blueprint::sparse::root_storage_tests` vs. `blueprint::sparse::root_storage_tests_smt`. I think the former is cleaner, but I'm fine with the other as well and since that's what we did for the merklized tests I'll do the same here.

Reply:
Done in https://github.com/FuelLabs/fuel-core/pull/2914/commits/5422fde10d363e5dc344a3c0f999c849185901fc